README.txt

Look at http://de.wikipedia.org/wiki/Datenstandards_zur_Beschreibung_des_Geschlechts
for details on the standards.
Some Extensions of TYPO3 use values for genders that don't fit to any standard.


ChangeLog
2009-09-04 David Bruchmann  <agentur@bruchmann-web.de>

	* initial code generated with kickstarter


tca.php
<?php
if (!defined ('TYPO3_MODE')) 	die ('Access denied.');

$TCA['static_sex'] = array (
	'ctrl' => $TCA['static_sex']['ctrl'],
	'interface' => array (
		'showRecordFieldList' => 'sys_language_uid,l10n_parent,l10n_diffsource,title,title_de,description,description_de,iso5218,active_entry,cdc,ecml,naaccr,usa_census,hl7,x12,astm_e1633,dcm,zims,obis,ubif,ubif_short'
	),
	'feInterface' => $TCA['static_sex']['feInterface'],
	'columns' => array (
		'sys_language_uid' => array (		
			'exclude' => 1,
			'label'  => 'LLL:EXT:lang/locallang_general.xml:LGL.language',
			'config' => array (
				'type'                => 'select',
				'foreign_table'       => 'sys_language',
				'foreign_table_where' => 'ORDER BY sys_language.title',
				'items' => array(
					array('LLL:EXT:lang/locallang_general.xml:LGL.allLanguages', -1),
					array('LLL:EXT:lang/locallang_general.xml:LGL.default_value', 0)
				)
			)
		),
		'l10n_parent' => array (		
			'displayCond' => 'FIELD:sys_language_uid:>:0',
			'exclude'     => 1,
			'label'       => 'LLL:EXT:lang/locallang_general.xml:LGL.l18n_parent',
			'config'      => array (
				'type'  => 'select',
				'items' => array (
					array('', 0),
				),
				'foreign_table'       => 'static_sex',
				'foreign_table_where' => 'AND static_sex.pid=###CURRENT_PID### AND static_sex.sys_language_uid IN (-1,0)',
			)
		),
		'l10n_diffsource' => array (		
			'config' => array (
				'type' => 'passthrough'
			)
		),
		'title' => array (		
			'exclude' => 0,		
			'label' => 'LLL:EXT:mdb_static_sex/locallang_db.xml:tx_mdbstaticsex_sex.title',		
			'config' => array (
				'type' => 'input',	
				'size' => '30',	
				'eval' => 'required,trim',
			)
		),
		'title_de' => array (		
			'exclude' => 0,		
			'label' => 'LLL:EXT:mdb_static_sex/locallang_db.xml:tx_mdbstaticsex_sex.title_de',		
			'config' => array (
				'type' => 'input',	
				'size' => '30',
			)
		),
		'description' => array (		
			'exclude' => 0,		
			'label' => 'LLL:EXT:mdb_static_sex/locallang_db.xml:tx_mdbstaticsex_sex.description',		
			'config' => array (
				'type' => 'text',
				'cols' => '30',	
				'rows' => '5',
			)
		),
		'description_de' => array (		
			'exclude' => 0,		
			'label' => 'LLL:EXT:mdb_static_sex/locallang_db.xml:tx_mdbstaticsex_sex.description_de',		
			'config' => array (
				'type' => 'text',
				'cols' => '30',	
				'rows' => '5',
			)
		),
		'iso5218' => array (		
			'exclude' => 0,		
			'label' => 'LLL:EXT:mdb_static_sex/locallang_db.xml:tx_mdbstaticsex_sex.iso5218',		
			'config' => array (
				'type'     => 'input',
				'size'     => '2',
				'max'      => '1',
				'eval'     => 'int',
				'checkbox' => '',
				'range'    => array (
					'upper' => '9',
					'lower' => '0'
				),
				'default' => 0
			)
			
		),
		'active_entry' => array (		
			'exclude' => 0,		
			'label' => 'LLL:EXT:mdb_static_sex/locallang_db.xml:tx_mdbstaticsex_sex.active_entry',		
			'config' => array (
				'type'     => 'input',
				'size'     => '2',
				'max'      => '1',
				'eval'     => 'int',
				'checkbox' => '',
				'range'    => array (
					'upper' => '9',
					'lower' => '0'
				),
				'default' => 0
			)
		),
		'cdc' => array (		
			'exclude' => 0,		
			'label' => 'LLL:EXT:mdb_static_sex/locallang_db.xml:tx_mdbstaticsex_sex.cdc',		
			'config' => array (
				'type' => 'input',	
				'size' => '2',	
				'max' => '1',	
				'eval' => 'trim',
			)
		),
		'ecml' => array (		
			'exclude' => 0,		
			'label' => 'LLL:EXT:mdb_static_sex/locallang_db.xml:tx_mdbstaticsex_sex.ecml',		
			'config' => array (
				'type' => 'input',	
				'size' => '2',	
				'max' => '1',	
				'eval' => 'trim',
			)
		),
		'naaccr' => array (		
			'exclude' => 0,		
			'label' => 'LLL:EXT:mdb_static_sex/locallang_db.xml:tx_mdbstaticsex_sex.naaccr',		
			'config' => array (
				'type'     => 'input',
				'size'     => '2',
				'max'      => '1',
				'eval'     => 'int',
				'checkbox' => '',
				'range'    => array (
					'upper' => '9',
					'lower' => '0'
				),
				'default' => 0
			)
		),
		'usa_census' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:mdb_static_sex/locallang_db.xml:tx_mdbstaticsex_sex.usa_census',		
			'config' => array (
				'type'     => 'input',
				'size'     => '2',
				'max'      => '1',
				'eval'     => 'int',
				'checkbox' => '',
				'range'    => array (
					'upper' => '2',
					'lower' => '1'
				),
				'default' => 0
			)
		),
		'hl7' => array (		
			'exclude' => 0,		
			'label' => 'LLL:EXT:mdb_static_sex/locallang_db.xml:tx_mdbstaticsex_sex.hl7',		
			'config' => array (
				'type' => 'input',	
				'size' => '2',	
				'max' => '1',	
				'eval' => 'trim',
			)
		),
		'x12' => array (		
			'exclude' => 0,		
			'label' => 'LLL:EXT:mdb_static_sex/locallang_db.xml:tx_mdbstaticsex_sex.x12',		
			'config' => array (
				'type' => 'input',	
				'size' => '2',	
				'max' => '1',	
				'eval' => 'trim',
			)
		),
		'astm_e1633' => array (		
			'exclude' => 0,		
			'label' => 'LLL:EXT:mdb_static_sex/locallang_db.xml:tx_mdbstaticsex_sex.astm_e1633',		
			'config' => array (
				'type' => 'input',	
				'size' => '3',	
				'max' => '2',	
				'eval' => 'trim',
			)
		),
		'dcm' => array (		
			'exclude' => 0,		
			'label' => 'LLL:EXT:mdb_static_sex/locallang_db.xml:tx_mdbstaticsex_sex.dcm',		
			'config' => array (
				'type' => 'input',	
				'size' => '7',	
				'max' => '6',	
				'eval' => 'trim',
			)
		),
		'zims' => array (		
			'exclude' => 0,		
			'label' => 'LLL:EXT:mdb_static_sex/locallang_db.xml:tx_mdbstaticsex_sex.zims',		
			'config' => array (
				'type' => 'input',	
				'size' => '14',	
				'max' => '13',	
				'eval' => 'trim',
			)
		),
		'obis' => array (		
			'exclude' => 0,		
			'label' => 'LLL:EXT:mdb_static_sex/locallang_db.xml:tx_mdbstaticsex_sex.obis',		
			'config' => array (
				'type' => 'input',	
				'size' => '2',	
				'max' => '1',	
				'eval' => 'trim',
			)
		),
		'ubif' => array (		
			'exclude' => 0,		
			'label' => 'LLL:EXT:mdb_static_sex/locallang_db.xml:tx_mdbstaticsex_sex.ubif',		
			'config' => array (
				'type' => 'input',	
				'size' => '30',	
				'max' => '30',	
				'eval' => 'trim',
			)
		),
		'ubif_short' => array (		
			'exclude' => 0,		
			'label' => 'LLL:EXT:mdb_static_sex/locallang_db.xml:tx_mdbstaticsex_sex.ubif_short',		
			'config' => array (
				'type' => 'input',	
				'size' => '3',
				'max' => '2',	
			)
		),
	),
	'types' => array (
		'0' => array('showitem' => 'sys_language_uid;;;;1-1-1, l10n_parent, l10n_diffsource, title;;;;2-2-2, title_de;;;;3-3-3, description, description_de, iso5218, active_entry, cdc, ecml, naaccr, usa_census, hl7, x12, astm_e1633, dcm, zims, obis, ubif, ubif_short')
	),
	'palettes' => array (
		'1' => array('showitem' => '')
	)
);
?>

ext_tables.php
<?php
if (!defined ('TYPO3_MODE')) {
	die ('Access denied.');
}
$TCA['static_sex'] = array (
	'ctrl' => array (
		'title'     => 'LLL:EXT:mdb_static_sex/locallang_db.xml:tx_mdbstaticsex_sex',
		'readOnly' => 1, // Prevents the table from being altered
		//'adminOnly' => 1, // Only admin, if any
		'rootLevel' => 1,
		'is_static' => 1,
		'label'     => 'title',	
		'tstamp'    => 'tstamp',
		'crdate'    => 'crdate',
		'cruser_id' => 'cruser_id',
		'languageField'            => 'sys_language_uid',	
		'transOrigPointerField'    => 'l10n_parent',	
		'transOrigDiffSourceField' => 'l10n_diffsource',	
		'default_sortby' => 'ORDER BY crdate',
		'dynamicConfigFile' => t3lib_extMgm::extPath('mdb_static_sex').'tca.php',
		'iconfile'          => t3lib_extMgm::extRelPath('mdb_static_sex').'icon_tx_mdbstaticsex_sex.png',
	),
);
?>

ext_emconf.php
<?php

########################################################################
# Extension Manager/Repository config file for ext: "mdb_static_sex"
#
# Auto generated 21-12-2009 04:28
#
# Manual updates:
# Only the data in the array - anything else is removed by next write.
# "version" and "dependencies" must not be touched!
########################################################################

$EM_CONF[$_EXTKEY] = array(
	'title' => 'Static Sex',
	'description' => 'Different Standards to describe the sexual gender for human and zoological purposes.
Infos from http://de.wikipedia.org/wiki/Datenstandards_zur_Beschreibung_des_Geschlechts',
	'category' => '',
	'author' => 'David Bruchmann',
	'author_email' => 'mdb_static_sex@bruchmann-web.de',
	'shy' => '',
	'dependencies' => '',
	'conflicts' => '',
	'priority' => '',
	'module' => '',
	'state' => 'alpha',
	'internal' => '',
	'uploadfolder' => 0,
	'createDirs' => '',
	'modify_tables' => '',
	'clearCacheOnLoad' => 1,
	'lockType' => '',
	'author_company' => '',
	'version' => '0.0.2',
	'constraints' => array(
		'depends' => array(
		),
		'conflicts' => array(
		),
		'suggests' => array(
		),
	),
	'suggests' => array(
	),
	'_md5_values_when_last_written' => 'a:12:{s:9:"ChangeLog";s:4:"2d01";s:10:"README.txt";s:4:"5ce0";s:12:"ext_icon.gif";s:4:"cdc4";s:14:"ext_tables.php";s:4:"628e";s:14:"ext_tables.sql";s:4:"39b7";s:25:"ext_tables_static+adt.sql";s:4:"97bd";s:28:"icon_tx_mdbstaticsex_sex.png";s:4:"bf03";s:25:"icon_tx_staticsex_sex.png";s:4:"bf03";s:16:"locallang_db.xml";s:4:"6282";s:7:"tca.php";s:4:"b22f";s:19:"doc/wizard_form.dat";s:4:"3d5d";s:20:"doc/wizard_form.html";s:4:"38a9";}',
);

?>